File: YardyClient/src/app/models/UserProfile.ts
export class UserProfile {
  firstName: string;
  lastName: string;
  email: string;
  dob: string;
  street: string;
  city: string;
  zip: string;

  constructor(response: any)
  {
    this.firstName = response.firstName;
    this.lastName = response.lastName;
    this.email = response.email;
    this.dob = response.dob;
    this.street = response.street;
    this.city = response.city;
    this.zip = response.zip;
  }
}

File: Yardy/Models/UserProfile.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace Yardy.Models
{
    public class UserProfile
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public string Firstname { get; set; }
        [Required]
        public string Lastname { get; set; }
        [EmailAddress]
        [Required]
        public string Email { get; set; }
        

        public DateTime Dob { get; set; }
        
        public string Street { get; set; }
        
        public string City { get; set; }
        
        public string PostalCode { get; set; }

        public DateTime Created { get; set; }
    }
}